scripts/validate-artifacts.ps1

```diff
@@ -34,122 +34,93 @@ Write-Output "Validating artifacts"
 $TemplateFileText = [System.IO.File]::ReadAllText("$tempFolderPath/$resourcesTemplateName")
 $TemplateObject = ConvertFrom-Json $TemplateFileText -AsHashtable
 
-$nsg = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups")
-if ($nsg) {
-    if ($nsg.name.Count -eq 1) { 
-        Write-Output "`u{2705} Checked if the Network Security Group resource exists - OK"
+$virtualMachine = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Compute/virtualMachines" )
```

Comment:
Ensure that the `$TemplateObject.resources` array contains the expected resource types and properties. If the JSON structure changes, this line may need to be updated.